### PR TITLE
Chip-id in port selection (VSC-1359)

### PR DESCRIPTION
**Pull request by Hahihula**
[## Description

Switcxhed serial port listing to node serialport and providing additional information about connected board chip ( if the board is ESP).

## Type of change

- New feature (non-breaking change which adds functionality)

## Steps to test this pull request

Provide a list of steps to test changes in this PR and required output
Click on "Select port to use"
Wait for ~2 seconds
Observe results.

- Expected behaviour:
image

- Expected output:

## How has this been tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- Test A
- Test B

**Test Configuration**:
* ESP-IDF Version:
* OS (Windows,Linux and macOS):

## Dependent components impacted by this PR:

- Component 1
- Component 2

## Checklist
- [ ] PR Self Reviewed
- [ ] Applied Code formatting
- [ ] Added Documentation
- [ ] Added Unit Test
- [ ] Verified on all platforms - Windows,Linux and macOS

### CODE REVIEW

**Review by kolipakakondal — APPROVED** [review]
LGTM

**Comment by kolipakakondal** [comment]
Hi @AndriiFilippov Please verify this

File: src/espIdf/serial/serialPort.ts [hunk 1]
```diff
@@ -87,24 +86,50 @@ export class SerialPort {
...
+          workspaceFolder
+        );
+        const esptoolPath = join(
+          idfPath,
+          "components",
+          "esptool_py",
+          "esptool",
+          "esptool.py"
+        );
+        async function processPorts(serialPort: SerialPortDetails) {
```
Comment (brianignacio5) [inline src/espIdf/serial/serialPort.ts:114]: Could we add validation that esptool.py exists ?

File: src/espIdf/serial/serialPort.ts [hunk 2]
```diff
@@ -87,24 +86,50 @@ export class SerialPort {
...
 
-        if (arrayPrint) {
-          arrayPrint.forEach((portStr) => {
-            const portChoice = portStr.replace(/'/g, "").trim();
-            choices.push(new SerialPortDetails(portChoice));
-          });
-          resolve(choices);
-        } else {
-          reject(new Error("No serial ports found"));
+            serialPort.chipType = chipIdString2[1].trim();
```
Comment (brianignacio5) [inline src/espIdf/serial/serialPort.ts:125]: Could we add validation that match exists and it has at least 2 elements here ?

**Review by brianignacio5 — CHANGES_REQUESTED** [review]
Few validations I think would be good but otherwise LGTM

File: src/espIdf/serial/serialPort.ts (hunk 1 above)
Comment (Hahihula) [reply src/espIdf/serial/serialPort.ts:114]: ok, added

File: src/espIdf/serial/serialPort.ts (hunk 2 above)
Comment (Hahihula) [reply src/espIdf/serial/serialPort.ts:125]: good idea!

**Comment by AndriiFilippov** [comment]
@kolipakakondal @brianignacio5 hi ! 

Tested under: 
OS - Windows 10 
ESP-IDF: v5.2.1 

Do see chip info while multiple boards are connected. 

image

**Review by brianignacio5 — APPROVED** [review]
LGTM

**Pull request merged**